internal_types/src/schema.rs

```diff
@@ -218,6 +218,39 @@ impl Schema {
         Ok(record)
     }
 
+    /// Set the order of sort columns to the primary key id the schema
+    pub fn set_sort_key(&mut self, sort_key: &SortKey<'_>) {
+        let fields = self.inner.fields();
+
+        // create a new_fields that are the fields with their sort keys set
+        let new_fields = fields
+            .iter()
+            .map(|field| {
+                let mut new_field = field.clone();
+                if let Some(sort) = sort_key.get(field.name()) {
+                    let mut meta = std::collections::BTreeMap::new();
+                    // New sort key
+                    meta.insert(COLUMN_SORT_METADATA_KEY.to_string(), sort.to_string());
+                    // Keep other meta data
+                    if let Some(metadata) = field.metadata().to_owned() {
```

Comment:
I wonder if you have to do `to_owned()` here (which makes a copy) -- can you instead do something like

```suggestion
                    if let Some(metadata) = field.metadata() {
```

Reply:
Good catch. The to_owned definitely no need here.The original copy intention for different purpose but I did not use it and forgot to remove the to_owned